include/dev/hallSensor.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <EVT/io/GPIO.hpp>
+
+namespace IO = EVT::core::IO;
+
+namespace WSS::DEV {
+
+class hallSensor {
+
+public:
+    enum WheelSpeedState {
+        STOP = 0,    // First pulse or wheel is dead
+        MAINTAIN = 1,// Wheel is spinning at a constant speed or speeding up
+    };
+
+    // Constructor (take a GPIO instance and the radius of the wheel)
+    hallSensor(IO::GPIO& gpio, uint32_t wheelRadius);
+
+    uint32_t update();// Update the wheel speed
+
+    // Get the current speed of the wheel
+    uint32_t getSpeed(uint32_t timeDiff);
+
+private:
+    IO::GPIO& gpio;       // GPIO instance for the sensor
```

Comment:
This is how we one-line comment instance variables
```suggestion
    /** GPIO instance for the sensor */
    IO::GPIO& gpio;
```